src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationContext.cs

```diff
@@ -98,6 +98,43 @@ public ValidationContext(object instance, IServiceProvider? serviceProvider, IDi
             ObjectInstance = instance;
         }
 
+        /// <summary>
+        ///     Construct a <see cref="ValidationContext" /> for a given object instance with
+        ///     a <paramref name="displayName" />, an optional <paramref name="serviceProvider" />,
+        ///     and an optional property bag of <paramref name="items" />.
+        /// </summary>
+        /// <param name="instance">The object instance being validated.  It cannot be null.</param>
+        /// <param name="displayName">The display name associated with the object instance.</param>
+        /// <param name="serviceProvider">
+        ///     Optional <see cref="IServiceProvider" /> to use when <see cref="GetService" /> is called.
+        ///     If it is null, <see cref="GetService" /> will always return null.
+        /// </param>
+        /// <param name="items">
+        ///     Optional set of key/value pairs to make available to consumers via <see cref="Items" />.
+        ///     If null, an empty dictionary will be created.  If not null, the set of key/value pairs will be copied into a
+        ///     new dictionary, preventing consumers from modifying the original dictionary.
+        /// </param>
+        /// <exception cref="ArgumentNullException">When <paramref name="instance" /> is <c>null</c></exception>
+        /// <remarks>
+        ///     This constructor is trim-safe because it does not use reflection to resolve
+        ///     the Type of the <paramref name="instance" /> to support setting the DisplayName.
+        /// </remarks>
+        public ValidationContext(object instance, string displayName, IServiceProvider? serviceProvider, IDictionary<object, object?>? items)
```

Comment:
Which part of the other constructor forces it to have an RUC annotation? This seems to be replicating most of the statements present in the previous ctor.

Reply:
This code is the reason why the other 2 constructors are marked RUC:

https://github.com/dotnet/runtime/blob/5ff417fd7fbff9bbd3c66f35a1721ddaa37840eb/src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationContext.cs#L181-L182

Reply:
I updated the RUC messages here to be a little bit clearer about the fact that the issue is not providing a display name forces us down the path of resolving the type from the `object instance` which is not trim-safe.

Reply:
Pretty wild that getting the display name was the pitfall here. Very cool to overcome that, @captainsafia!